python/GUARDDUTY_UNTREATED_FINDINGS/GUARDDUTY_UNTREATED_FINDINGS.py

```diff
@@ -0,0 +1,509 @@
+    Then: set Default Values and evaluate
+
+  Scenario: 3
+    Given: No untreated GuardDuty findings are present.
+    Then: Return COMPLIANT
+
+  Scenario: 4
+    Given: GuardDuty has untreated Low/Medium/High Severity findings.
+    And: Low/Medium/High Severity findings are older than daysLowSev/daysMediumSev/daysHighSev number of days respectively.
+    Then: Return NON_COMPLIANT
+
+  Scenario: 5
+    Given: GuardDuty has untreated Low/Medium/High Severity findings.
+    And: Low/Medium/High Severity findings are not older than daysLowSev/daysMediumSev/daysHighSev number of days respectively.
+    Then: Return COMPLIANT
+"""
+
+import json
+import sys
+import datetime
+from datetime import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    evaluations = []
+    guardduty_client = get_client('guardduty', event)
+
+    #check whether GuardDutry is enabled or Not.
+    guardduty_detector_list = guardduty_client.list_detectors()
+    if not guardduty_detector_list['DetectorIds']:
+        return None
+
+    # Guard Duty supports only one detector per region, hence retrieving first Id.
+    guardduty_detector_id = guardduty_detector_list['DetectorIds'][0]
+
+    # check whether GuardDuty Suspended or Not. if detector is disabled then return NOT_APPLICABLE
+    guardduty_detector_status = guardduty_client.get_detector(
+        DetectorId=guardduty_detector_id
+    )
+    if guardduty_detector_status['Status'] == 'DISABLED':
+        return None
+
+    # get all the guardduty findings
+    all_findings_list = list_all_findings(guardduty_client, guardduty_detector_id)
+
+    # If no findings present then return COMPLIANT
+    if not all_findings_list:
+        return build_evaluation(event['accountId'], 'COMPLIANT', event)
+
+    # evaluate each finding
+    for each_finding in all_findings_list:
+        days_since_created = get_delta_days(each_finding['CreatedAt'])
+
+        # 0.1 to 3.9 severity score is LowSev
+        if 0.1 <= each_finding['Severity'] <= 3.9:
+            if days_since_created <= valid_rule_parameters['daysLowSev']:
+                evaluations.append(build_evaluation(each_finding['Id'], 'COMPLIANT', event))
+                continue
+            evaluations.append(build_evaluation(each_finding['Id'], 'NON_COMPLIANT', event, annotation='This AWS GurdDuty Low Severity finding is older than {} days'.format(valid_rule_parameters['daysLowSev'])))

```

Comment:
Please add the final dot after "days" word in both the annotation.

Reply:
Done.